packages/runtime-vapor/src/apiCreateSlots.ts

```diff
@@ -0,0 +1,61 @@
+// NOTE: this filed is based on `runtime-core/src/helpers/createSlots.ts`
+
+import { isArray } from '@vue/shared'
+import type { InternalSlots, Slot } from './componentSlots'
+import { renderEffect } from './renderEffect'
+
+// TODO: SSR
+
+interface CompiledSlotDescriptor {
+  name: string
+  fn: Slot
+  key?: string
+}
+
+export const createSlots = (
+  slots: InternalSlots,
+  dynamicSlotsGetter?: () => (
+    | CompiledSlotDescriptor
+    | CompiledSlotDescriptor[]
+    | undefined
+  )[],
+): InternalSlots => {
+  const dynamicSlotKeys: Record<string, true> = {}
+  renderEffect(() => {
```

Comment:
Not `renderEffect`
https://github.com/vuejs/core-vapor/blob/5c9a15140d1f4aa72706f2b7a4b72afaa537c0e3/packages/runtime-vapor/src/componentProps.ts#L125-L127